### PR TITLE
Add delete aliases API to the high-level REST client

### PR DESCRIPTION
**Pull request by DaveCTurner**
Relates #27205.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/alias/DeleteAliasesRequest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * A request to delete aliases from one or more indices.
+ */
+public class DeleteAliasesRequest extends AcknowledgedRequest<IndicesAliasesRequest> {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/admin/indices/alias/DeleteAliasesRequest.java]: The pluralisation is a bit strange - why not `DeleteAliasRequest`?  We already have `GetAliasesRequest` and `IndicesAliasesReqest` so I followed that.

**Comment by DaveCTurner** [comment]
I think the build failed for unrelated reasons that are fixed in `master`, but the intake build has gone red again. I merged the last green commit into this branch.

**Comment by DaveCTurner** [comment]
@hub-cap, @jdconrad, I wouldn't be surprised to hear that this needs more work to bring it in line with the new plans for the structure of the HLRC, but I've not been keeping on top of how that is developing so I'm not sure what needs doing. It'd be good if you could have a brief look at this and let me know what needs moving around.

**Comment by hub-cap** [comment]
hey sorry @DaveCTurner WRT not reviewing faster. If you *want* to update it you can, but I have not severed the tie between client and server, so you can technically leave all server things (not xpack related) in server and just reference them. Its going to be a big ole effort to move them, and you dont have to pave that road with this PR :)

**Review by jdconrad — APPROVED** [review]
@DaveCTurner Sorry for not getting to this sooner.  LGTM.  As long as @hub-cap is fine with this not supporting the upcoming changes.

**Review by hub-cap — CHANGES_REQUESTED** [review]
My only question here is on whether we should be doing the request builder style, or if we should be doing the validation and assignment at constructor time and making them final. Since they are not ser/deser via the readFrom/writeTo we should try to make them final. Your `DeleteAliasesRequest` actually does not need to override those methods, as the new style classes will not need to implement them.  Also, you can feel free to use `List`'s instead of `String[]`.

**Comment by javanna** [comment]
does this need to be updated or can it be closed?

**Comment by DaveCTurner** [comment]
Sorry, this slipped right off my list. This API is not supported today in the HLRC (or the transport client) along with most of the other REST-only APIs, but I'm not going to have time to bring this PR back up to date in the near future, so it's probably best to close this.

**Pull request closed without merging**